src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -1727,12 +1791,444 @@ def get_no_wait(self, **kwargs) -> bool:
+                        load_balancer_profile.managed_outbound_i_ps or
+                        load_balancer_profile.outbound_i_ps or
+                        load_balancer_profile.outbound_ip_prefixes
+                    ):
+                        raise MutuallyExclusiveArgumentError(
+                            "userDefinedRouting doesn't support customizing a standard load balancer with IP addresses"
+                        )
+                else:
+                    if (
+                        self.get_load_balancer_managed_outbound_ip_count() or
+                        self.get_load_balancer_outbound_ips() or
+                        self.get_load_balancer_outbound_ip_prefixes()
+                    ):
+                        raise MutuallyExclusiveArgumentError(
+                            "userDefinedRouting doesn't support customizing a standard load balancer with IP addresses"
+                        )
+
+        return outbound_type
+
+    # pylint: disable=unused-argument
+    def get_network_plugin(self, enable_validation: bool = False, **kwargs) -> Union[str, None]:
+        # read the original value passed by the command
+        raw_value = self.raw_param.get("network_plugin")
+        # try to read the property value corresponding to the parameter from the `mc` object
+        value_obtained_from_mc = None
+        if self.mc and self.mc.network_profile:
+            value_obtained_from_mc = self.mc.network_profile.network_plugin
+
+        # set default value
+        if value_obtained_from_mc is not None:
+            network_plugin = value_obtained_from_mc
+        else:
+            network_plugin = raw_value
+
+        # this parameter does not need dynamic completion
+
+        # validation
+        if enable_validation:
+            if network_plugin:
+                if network_plugin == "azure" and self.get_pod_cidr():
+                    raise MutuallyExclusiveArgumentError(
+                        "Please use kubenet as the network plugin type when pod_cidr is specified"
+                    )
+            else:
+                if (
+                    self.get_pod_cidr() or
+                    self.get_service_cidr() or
+                    self.get_dns_service_ip() or
+                    self.get_docker_bridge_address() or
+                    self.get_network_policy()
+                ):
+                    raise RequiredArgumentMissingError(
+                        "Please explicitly specify the network plugin type"
+                    )
+        return network_plugin
+
+    # pylint: disable=unused-argument
+    def get_pod_cidr(self, enable_validation: bool = False, **kwargs) -> Union[str, None]:
+        # read the original value passed by the command
+        raw_value = self.raw_param.get("pod_cidr")
+        # try to read the property value corresponding to the parameter from the `mc` object
+        value_obtained_from_mc = None
+        if self.mc and self.mc.network_profile:
```

Comment:
here too.

Reply:
Changed in commit [524654c](https://github.com/Azure/azure-cli/pull/19458/commits/524654cc0c83db5b206ea2a30ec321e6632c8980).